src/lib/fcitx/inputcontext.cpp

```diff
@@ -256,7 +256,11 @@ void InputContext::updatePreedit() {
 void InputContext::updateUserInterface(UserInterfaceComponent component,
                                        bool immediate) {
     FCITX_D();
-    d->emplaceEvent<InputContextUpdateUIEvent>(component, this, immediate);
```

Comment:
I think we should move this part of the code to UserInterfaceManager::flush. Thus the event is always generated.